src/client/dfs/duns.c

```diff
@@ -852,6 +852,51 @@ duns_create_lustre_path(daos_handle_t poh, daos_pool_info_t info, const char *pa
 err:
 	return rc;
 }
+
+static int
+duns_link_lustre_path(const char *pool, const char *cont, daos_cont_layout_t type,
+		      const char *path, mode_t mode)
+{
+	char			str[DUNS_MAX_XATTR_LEN + 1];
+	int			len;
+	int			rc, rc2;
+
+	/* XXX if liblustreapi is not binded, do it now ! */
+	if (liblustre_binded == false && liblustre_notfound == false) {
+		rc = bind_liblustre();
+		if (rc)
+			return EINVAL;
+	}
+
+	/** create file/dir and store the daos attributes in the path LOV/LMV */
+	len = snprintf(str, DUNS_MAX_XATTR_LEN, DUNS_LUSTRE_XATTR_FMT, pool, cont);
+	if (len < 0) {
+		D_ERROR("Failed to create foreign LOV/LMV value\n");
+		return EINVAL;
+	}
+
+	if (type == DAOS_PROP_CO_LAYOUT_POSIX) {
+		rc = (*dir_create_foreign)(path, mode, LU_FOREIGN_TYPE_SYMLINK, (__u32)type, str);
+		if (rc) {
+			D_ERROR("Failed to create Lustre dir '%s' with foreign "
+				"LMV '%s' (rc = %d).\n", path, str, rc);
```

Comment:
Should we format the rc similar to this?
https://github.com/daos-stack/daos/blob/1e654aa4c1086d66db3e1339007a2ae9ab0e4575/src/client/dfs/duns.c#L570

Reply:
this is just copy paste from above.. probably need to do a wider scan of errno rc formatting in this file and others independent of this PR.